junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala

```diff
@@ -130,55 +130,94 @@ abstract class JUnitTest {
+          case (newMatcher, Nil)       => newMatcher
+          case (newMatcher, remaining) => newMatcher.matchUnordered(remaining)
         }
       }
-  
-      val (expected1, actual1, droppedFront) =
-        minimizeDiff(expected, actual, 0)
-      val (expected2, actual2, droppedBack) =
-        minimizeDiff(expected1.reverse, actual1.reverse, 0)
-      val expectedMinimized = expected2.reverse
-      val actualMinimized = actual2.reverse
-  
-      if (expectedMinimized != actualMinimized) {
-        val msg = new StringBuilder
-        def appendElems(original: List[Output], minimized: List[Output]): Unit = {
-          def appendGreyLine(out: Output): Unit = {
-            msg.append(GREY + "  " + withoutColor(out.toString) + NORMAL + "\n")
+
+      def assertEmpty(): Unit = assert(remaining.isEmpty)
+
+      /** Tries to match the prefix. If succeeds, returns a new matcher. */
+      private def tryMatch(prefix: List[Output]): Option[Matcher] = {
+        if (remaining.startsWith(prefix)) {
+          val (justMatched, newRemaining) = remaining.splitAt(prefix.length)
+          Some(new Matcher(matched ++ justMatched, newRemaining))
+        } else {
+          None
+        }
+      }
+
+      private def matchOneOf(
+          prefixes: List[List[Output]]): (Matcher, List[List[Output]]) = {
+        @tailrec
+        def loop(matcher: Matcher, toSearch: List[List[Output]],
+            tried: List[List[Output]]): (Matcher, List[List[Output]]) = {
+          toSearch match {
+            case x :: xs =>
+              matcher.tryMatch(x) match {
+                case Some(m) => (m, xs ::: tried)
+                case None    => loop(matcher, xs, x :: tried)
+              }
+
+            case Nil =>
+              matcher.fail(tried)
           }
-          original.slice(droppedFront - 3, droppedFront).foreach(appendGreyLine)
-          minimized.foreach(out => msg.append("  " + out + "\n"))
-          val backIndex = original.size - droppedBack
-          original.slice(backIndex, backIndex + 3).foreach(appendGreyLine)
         }
-  
+
+        loop(this, prefixes, Nil)
+      }
+
+      private def fail(expecteds: List[List[Output]]): Nothing = {
+        val msg = new StringBuilder
         msg.append(s"JUnit output mismatch with $argInfo:\n")
-        msg.append(s"Expected: List(\n")
-        appendElems(expected, expectedMinimized)
-        msg.append(")\nbut got: List(\n")
-        appendElems(actual, actualMinimized)
+        msg.append("Expected next, one of:\n")
+
+        def appendOut(out: Output) =
+          msg.append("  " + out + "\n")
+
+        for (expected <- expecteds) {
+          msg.append("List(\n")
+          expected.foreach(appendOut)
+          msg.append(")\n\n")
+        }
+
+        msg.append("but got: List(\n")
+
+        val maxLen = expecteds.map(_.size).max
+
+        matched.takeRight(3).foreach(out =>
+          msg.append(GREY + "  " + withoutColor(out.toString) + NORMAL + "\n"))
```

Comment:
Consider a for:
```scala
for (out <- matched.takeRight(3))
  msg.append(GREY + "  " + withoutColor(out.toString) + NORMAL + "\n")
```
According to https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#method-call-style, the `()`+`=>` association only works if it fits entirely on the same line.